main.go

```diff
@@ -147,6 +157,10 @@ func main() {
 
 	setupChecks(mgr)
 
+	if webhookPort != 0 {
```

Comment:
Why do we need this?

Reply:
We need a way to disable webhook somehow(e.q. local runs) and cluster-api-provider-metal3 uses this way.

Reply:
Not sure if you have used kubebuilder or not for scaffolding the webhook, but wouldn't be clear to have something similar [approach](https://book.kubebuilder.io/cronjob-tutorial/main-revisited.html)?

Reply:
Good point @andfasano, that usage seems more clear. I'll change that way.